### PR TITLE
Fix your portfolio screen bugs

### PR DESCRIPTION
**Pull request by ivanng-oursky**
- show connect wallet dialog if not logged-in
- incorrect balance calculation

### Screen

image

refs oursky/likedao#191

### CODE REVIEW

File: react-app/src/api/distributionAPI.ts [hunk 1]
```diff
@@ -67,16 +72,57 @@ export const useDistributionAPI = (): IDistributionAPI => {
...
       amount: convertMinimalTokenToToken(rewardAmount),
     };
-  }, [coinMinimalDenom, wallet, query]);
+  }, [wallet, query]);
+
+  const getTotalCommission = useCallback(async () => {
+    if (wallet.status !== ConnectionStatus.Connected) {
+      throw new Error("Wallet not connected");
+    }
+    if (!wallet.account.address.includes("valoper")) {
```
Comment (hochiw) [inline react-app/src/api/distributionAPI.ts:84]: seems more suitable to decode the bech32 and compare the prefix a config value

File: react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts [hunk 2]
```diff
@@ -32,15 +34,23 @@ export function usePortfolioQuery(): PortfolioRequestState {
...
+        reward,
+        profile,
+      ] = await Promise.all([
+        cosmosAPI.getBalance(),
+        cosmosAPI.getStakedBalance(),
+        staking.getUnstakingAmount(wallet.account.address),
+        distribution.getTotalCommission(),
+        distribution.getTotalDelegationRewards(),
+        desmosQuery.getProfile(
+          translateAddress(wallet.account.address, "desmos")
```
Comment (hochiw) [inline react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts:51]: so many apis :sosad:

**Review by hochiw — CHANGES_REQUESTED** [review]

File: react-app/src/api/distributionAPI.ts (hunk 1 above)
Comment (ivanng-oursky) [reply react-app/src/api/distributionAPI.ts]: I took another look at this, turns out we need to translate `like1...` address to `likevaloper1...` first, then use the `likevaloper1...` address to query commissions. In this case, checking for bech32 prefix is nolonger needed.

**Comment by ivanng-oursky** [comment]
@hochiw update 🙇 

see diff: https://github.com/oursky/likedao/compare/a8e0e5453db8b099aa0e5933436ecd957715d189..6140e28b3154cecc2c8f2c2bbae7ffdfd4189968

File: react-app/src/api/distributionAPI.ts [hunk 3]
```diff
@@ -67,16 +73,51 @@ export const useDistributionAPI = (): IDistributionAPI => {
...
     };
-  }, [coinMinimalDenom, wallet, query]);
+  }, [wallet, query]);
+
+  const getTotalCommission = useCallback(async () => {
+    if (wallet.status !== ConnectionStatus.Connected) {
+      throw new Error("Wallet not connected");
+    }
+
+    const address = translateAddress(wallet.account.address, "likevaloper");
```
Comment (hochiw) [inline react-app/src/api/distributionAPI.ts:86]: we are going to opensource it and it's possible for others to use prefix other than `likevaloper`, could be `xvaloper` and `yvaloper`, let's put this into config

**Review by hochiw — CHANGES_REQUESTED** [review]

**Comment by ivanng-oursky** [comment]
@hochiw updated 🙇

File: react-app/src/components/PortfolioScreen/PortfolioScreen.tsx [hunk 4]
```diff
@@ -8,16 +8,20 @@ import {
...
   const requestState = usePortfolioQuery();
   const { translate } = useLocale();
+  const wallet = useWallet();
 
   useEffectOnce(
     () => {
-      if (isRequestStateError(requestState)) {
+      if (wallet.status === ConnectionStatus.Idle) {
+        wallet.openConnectWalletModal();
+      } else if (isRequestStateError(requestState)) {
```
Comment (hochiw) [inline react-app/src/components/PortfolioScreen/PortfolioScreen.tsx:24]: It automatically refreshes after the user connects to the wallet right?

**Pull request merged**